presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/ClientInfo.java

```diff
@@ -28,17 +28,14 @@ public class ClientInfo
 
     private final String testId;
     private final Optional<String> testName;
-    private final String sourceQueryName;
-    private final String suite;
-    private final String queryStage;
+    private String sourceQueryName;
```

Comment:
Would it be possible to restructure code to avoid removing `final`? Generally we are trying to avoid using mutable data structures if not strongly necessary. 
Ha  you considered introducing an additional class to store the properties you want to provide via an extension?

Reply:
it makes sense. introduced clientinfoFactory and keep final keyword